public/app/features/dashboard/components/Inspector/QueryInspector.tsx

```diff
@@ -47,6 +56,30 @@ export class QueryInspector extends PureComponent<Props, State> {
     appEvents.on(CoreEvents.dsRequestResponse, this.onDataSourceResponse);
     appEvents.on(CoreEvents.dsRequestError, this.onRequestError);
     this.props.panel.events.on(PanelEvents.refresh, this.onPanelRefresh);
+    this.updateQueryList();
+  }
+
+  componentDidUpdate(oldProps: Props) {
+    if (this.props.data !== oldProps.data) {
+      this.updateQueryList();
+    }
+  }
+
+  updateQueryList() {
+    const { data } = this.props;
+    const executedQueries: ExecutedQueryInfo[] = [];
+    if (data?.length) {
+      data.forEach((frame, idx) => {
```

Comment:
Nice, didn't know we had `forEach` on dataFrame 🎉 ⭐

Reply:
data is an array, so don't be too impressed ;)

do check [vectorator](https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/vector/FunctionalVector.ts#L45)... that will let you do all the functional things you want with a vector

Reply:
thanks, will check it out!